lib/data_hygiene/specialist_sector_tagger.rb

```diff
@@ -0,0 +1,77 @@
+class SpecialistSectorTagger
+  def self.process_from_csv(csv_location, logger: Logger.new(nil))
+    CSV.foreach(csv_location, headers: true) do |row|
+      new(
+        row.fetch("document_type"),
+        row.fetch("slug"),
+        row.fetch("tag"),
+        logger: logger,
+      ).process
+    end
+  end
+
+  def initialize(document_type,
+                 document_slug,
+                 specialist_sector_slug,
+                 logger: Logger.new(nil)
+                )
+    @document_type = document_type
+    @document_slug = document_slug
+    @specialist_sector_slug = specialist_sector_slug
+    @logger = logger
+  end
+
+  def process
+    if document.nil?
+      log "warning: failed to find #{@document_type} document with slug" +
+        " #{@document_slug} - skipping"
+      return
+    end
+
+    document.editions.each do |edition|
+      if edition.specialist_sectors.any? {|s| s.tag == @specialist_sector_slug }
+        log "skipping '#{document.slug}' edition" +
+          " #{edition.id}(#{edition.state})" +
+          "- already tagged to #{@specialist_sector_slug}"
+      else
+        log "tagging '#{document.slug}' edition" +
```

Comment:
I think this logging could do with having the document_type logged too, since the slug is only unique within a type.